src/wallet/wallet2.cpp

```diff
@@ -8414,8 +8414,9 @@ bool wallet2::check_spend_proof(const crypto::hash &txid, const std::string &mes
   }
   std::vector<std::vector<crypto::signature>> signatures = { std::vector<crypto::signature>(1) };
   const size_t sig_len = tools::base58::encode(std::string((const char *)&signatures[0][0], sizeof(crypto::signature))).size();
-  THROW_WALLET_EXCEPTION_IF(sig_str.size() != header_len + num_sigs * sig_len,
-    error::wallet_internal_error, "incorrect signature size");
+  if( sig_str.size() != header_len + num_sigs * sig_len ) {
+    return false;
```

Comment:
I'm not sure what a MERROR is. This way I think it will just return "bad signature", which is what you want if you're checking a spendproof. You don't want an error on the signature not being good. There is no error here, unless you consider a bad signature to be an error, since if you check the signature against a different txid than the one used to create the signature, it would previously throw an error.

Reply:
It's a log. Fair point about it being not an error, a MDEBUG would be fine.

Reply:
This is the only contribution I've made to the monero project, so I'm not sure how I would add an MDEBUG/MERROR. I'm happy to answer any questions about the issue though.

Reply:
Alright, forget about it then, it's OK as it is.